toolchain/check/convert.cpp

```diff
@@ -539,9 +539,20 @@ static auto ConvertStructToClass(Context& context, SemIR::StructType src_type,
                                  SemIR::InstId value_id,
                                  ConversionTarget target) -> SemIR::InstId {
   PendingBlock target_block(context);
+  auto& class_info = context.classes().Get(dest_type.class_id);
+  if (class_info.inheritance_kind == SemIR::Class::Abstract) {
+    CARBON_DIAGNOSTIC(ConstructionOfAbstractClass, Error,
+                      "Cannot construct instance of abstract class `{0}`. "
+                      "Consider using `partial` class type instead.",
```

Comment:
class type -> facet?

https://github.com/carbon-language/carbon-lang/blob/trunk/docs/design/classes.md#partial-facet

```suggestion
                      "Consider using `partial` facet type instead.",
```

Reply:
@josh11b I *think* that's out of date; we use "facet" to mean a value of facet type these days, and it seems confusing to also use it here. (It wouldn't be wrong, because every type is a facet, because `type` is a facet type, but it seems at least misleading.)

Reply:
I agree that we've changed how we use the word "facet" since https://github.com/carbon-language/carbon-lang/blob/trunk/docs/design/classes.md#partial-facet was written. I don't have a good replacement term at the moment. "partial class type" would be fine with me until we have that discussion, though should it be "partial-class type"?

Reply:
I'm trying to suggest specifically using the `partial` keyword here. I suppose we could use
```
                      "Cannot construct instance of abstract class. "
                      "Consider using `partial {0}` instead.",
```
Thoughts, preferences?

Reply:
I've switched to the `partial {0}` approach for now.

Reply:
I could go either way on this; I think the change is fine to keep.